rustls-libssl/src/cache.rs

```diff
@@ -232,6 +232,15 @@ impl ServerSessionStorage {
         Arc::new(SslSession::new(id, value, context, time_out))
     }
 
+    /// Return `None` if `sess` has the wrong context value.
+    fn filter_session_context(&self, sess: Arc<SslSession>) -> Option<Arc<SslSession>> {
+        if self.get_context() != sess.context {
+            None
+        } else {
+            Some(sess)
+        }
```

Comment:
ultra-nit: I think this reads clearer reversed, with the `Some` condition first:

```rust
        if self.get_context() == sess.context {
            Some(sess)
        } else {
            None
        }
```